### DISCUSSION TITLE
What does the mapper configuration means?

### DISCUSSION THREAD

**Discussion by KelvinChung2000** [tancheng/CGRA-Mapper#17] [Q&A]
So, for example, with the following:

```
  {
    "x"           : 2,
    "y"           : 1,
    "cycle"       : 0,
    "opt"         : "OPT_ADD_CONST",
    "predicate"   : 0,
    "out_0"       : "none",
    "out_1"       : "none",
    "out_2"       : "none",
    "out_3"       : "none",
    "out_4"       : "4",
    "out_5"       : "none",
    "out_6"       : "none",
    "out_7"       : "none"
  },
```

This configuration is mapped to the node at (2,1). `out_[0-3]` suggests it is not outputting any data, and `out_[4-7]` suggests it is taking only one input and the input is coming from the `east` side. `predicate=0` means it will always execute. Is my understanding correct?

For an operation like OPT_ADD_CONST, how do I obtain the constant that is being added? If this operation is the "loop counter", how can the configuration tell it needs to store the result and reuse it in the next cycle?

**Comment by tancheng** [comment]
Your understanding is exactly correct.

For the constants, there is actually a [constant queue](https://github.com/tancheng/VectorCGRA/blob/37456a3f15503e8de5d1a592330ff68bd249872b/cgra/test/CGRACL_FIR_demo_test.py#L148) in each tile. Data needs to be carefully preloaded into it.

For `iter++`, it is actually not single DFG node, it is performed by `phi-add-cmp-br` 4 nodes in a dependency cycle. So data is still flowing to other tile (or itself), usually 4 clock cycles needed.
